### ISSUE TITLE
Payment methods in payments title in wrong language

### ISSUE THREAD

**Issue by Bartlomiejsz** [magento/magento2#7582]
### Preconditions
1. Magento 2.1.2
2. Having at least two store views (e.g. EN and FR)
3. Payment method configured with different title for each store view (e.g. Invoice - EN, Facture - FR)

### Steps to reproduce
1. Add some product to cart in EN store view.
2. Go to checkout second step (Payment method title is correct - Invoice).
3. Switch to FR store view.
4. Go to second checkout step.

### Expected result
1. Payment method title should now be 'Facture'

### Actual result

1. The title is still 'Invoice', because of those code lines:
```
public function getAvailableMethods(\Magento\Quote\Api\Data\CartInterface $quote = null)
{
    $store = $quote ? $quote->getStoreId() : null;
    $methods = [];
    foreach ($this->paymentHelper->getStoreMethods($store, $quote) as $method) {
```

**Comment by vityakopin** [comment]
Thanks for reporting this issue. I cannot reproduce this. Try to delete cookies after switching between store views. Could you confirm that this issue is still actual?

**Comment by Bartlomiejsz** [comment]
@vityakopin @veloraven issue still persists on 2.1.7

**Comment by vityakopin** [comment]
Internal ticket MAGETWO-70060 was created.

**Comment by magento-engcom-team** [comment]
@Bartlomiejsz, thank you for your report.
We've created internal ticket(s) MAGETWO-70060, MAGETWO-73214 to track progress on the issue.

**Comment by bka** [comment]
Issue is still present on 2.2-develop. What you can observe is, that the store view from which you put product into the quote matters. If I put a product from other store view e.g. FR into cart and go to checkout the translation will be correct.

It's because the available payment methods and their titles are fetched via API call to _/rest/de/V1/carts/mine/payment-information_. This API call can only indirectly determine which store view to use. In `app/code/Magento/Payment/Model/MethodList.php` the storeId to use is coming from the quote.

    public function getAvailableMethods(\Magento\Quote\Api\Data\CartInterface $quote = null)
    {
        $store = $quote ? $quote->getStoreId() : null;

This quote object gets initialized in and overwritten by `Magento/Quote/Model/Quote::loadByCustomer`. StoreId is coming from quote in this case.

    protected function loadQuote($loadMethod, $loadField, $identifier, array $sharedStoreIds = [])
    {
        /** @var Quote $quote */
        $quote = $this->quoteFactory->create();
        if ($sharedStoreIds) {
            $quote->setSharedStoreIds($sharedStoreIds);
        }
        $quote->setStoreId($this->storeManager->getStore()->getId())->$loadMethod($identifier);
        if (!$quote->getId()) {
            throw NoSuchEntityException::singleField($loadField, $identifier);
        }
        return $quote;
    }

**Comment by okorshenko** [comment]
Hi @Bartlomiejsz the issue has been fixed in 2.2-develop branch and will be available in 2.2.1 release soon

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-80894

**Comment by magento-team** [comment]
Hi @Bartlomiejsz. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1282 by @magento-engcom-team in 2.3-develop branch
Related commit(s):


The fix will be available with the upcoming patch release.
